src/integration_tests/rpc_multiple_remote_addrs/main.cc

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2016 Alexander Gallego. All rights reserved.
+//
+// std
+#include <chrono>
+#include <iostream>
+// linux
+#include <arpa/inet.h>
+#include <ifaddrs.h>
+#include <net/if.h>
+// seastar
+#include <core/app-template.hh>
+#include <core/distributed.hh>
+#include <core/sleep.hh>
+#include <net/api.hh>
+// smf
+#include "integration_tests/non_root_port.h"
+#include "smf/log.h"
+#include "smf/random.h"
+#include "smf/rpc_generated.h"
+#include "smf/rpc_handle_router.h"
+#include "smf/rpc_recv_context.h"
+#include "smf/rpc_server.h"
+// templates
+#include "integration_tests/demo_service.smf.fb.h"
+
+// For a network connected host we should always have at least an lo and eth0 interface 
+// (or whatever the connected interface is called) ... so capping the maximum test clients
+// at 2 will cover this. Going larger doesn't improve the accuracy of this test.
+constexpr static const size_t kMaxClients = 2;
+
+struct client_info {
+    explicit client_info(seastar::ipv4_addr ip): client(smf_gen::demo::SmfStorageClient::make_shared(ip)), ip(ip.ip) {};
+    client_info(){};
+    seastar::shared_ptr<smf_gen::demo::SmfStorageClient> client = nullptr;
+    uint32_t ip = 0;
+    char ip_str[INET_ADDRSTRLEN]{};
+};
+
+class clients {
+public:
+    explicit clients(uint16_t port) {
+        struct ifaddrs *ifaddr;
+        LOG_THROW_IF(getifaddrs(&ifaddr) == -1, "getifaddrs()");
+        struct ifaddrs *ifa = clients::next_valid_interface(ifaddr);
+        for (int i = 0; i < kMaxClients; ifa = ifa->ifa_next, i++) {
+            ifa = clients::next_valid_interface(ifa);
+            THROW_IFNULL(ifa);
+            auto sa_in = reinterpret_cast<struct sockaddr_in*>(ifa->ifa_addr);
+            auto ip = ntohl(sa_in->sin_addr.s_addr);
+            clients_.push_back(client_info(seastar::make_ipv4_address(ip, port)));
+            THROW_IFNULL(inet_ntop(AF_INET, &sa_in->sin_addr, clients_[i].ip_str, INET_ADDRSTRLEN));
+            DLOG_DEBUG("Added client ip {}", clients_[i].ip_str);
+        }
+        freeifaddrs(ifaddr);
+    }
+
+    std::vector<struct client_info>& iterator() { return clients_; }
```

Comment:
this is good but the short form is 

`auto begin() { return clients_.begin(); }`
`auto end() { return clients_.end(); }`